### PR TITLE
Enable access to scim2 Role endpoint

### PR DESCRIPTION
**Pull request by ashensw**
Resolves wso2/product-is#8591

### CODE REVIEW

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 1]
```diff
@@ -1587,6 +1595,22 @@
             <Permissions>/permission/admin/manage/identity/rolemgt/delete</Permissions>
             <Scopes>internal_role_mgt_delete</Scopes>
         </Resource>
+        <Resource context="(.*)/scim2/Roles/(.*)" secured="true" http-method="GET">
```
Comment (mefarazath) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:1598]: Why not have all the "(.*)/scim2/Roles/(.*)" entries together?

Comment (ashensw) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:1598]: Groups config had the same way. Moved to one place and updated the PR.

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml [hunk 2]
```diff
@@ -1205,6 +1205,30 @@
...
+            <Scopes>internal_role_mgt_create</Scopes>
+        </Resource>
+        <Resource context="(.*)/scim2/Roles" secured="true" http-method="GET">
+            <Permissions>/permission/admin/manage/identity/rolemgt/view</Permissions>
+            <Scopes>internal_role_mgt_view</Scopes>
+        </Resource>
+        <Resource context="(.*)/scim2/Roles/(.*)" secured="true" http-method="GET">
+            <Permissions>/permission/admin/manage/identity/rolemgt/view</Permissions>
+            <Scopes>internal_role_mgt_view</Scopes>
+        </Resource>
```
Comment (mefarazath) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml:1219]: Can we combine these two by having the context as (.*)/scim2/Roles(.*)

Comment (ashensw) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml]: Fixed and updated the PR.

**Pull request merged**